cpp/include/cuspatial/experimental/spatial_window.cuh

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cuspatial/vec_2d.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+
+#include <thrust/iterator/iterator_traits.h>
+
+namespace cuspatial {
+
+/**
+ * @brief Count of points (x,y) that fall within a rectangular query window.
+ *
+ * @ingroup spatial_relationship
+ *
+ * A point (x, y) is in the window if `x > window_min_x && x < window_max_x && y > window_min_y &&
+ * y < window_max_y`.
+ *
+ * Swaps `window_min.x` and `window_max.x` if `window_min.x > window_max.x`.
+ * Swaps `window_min.y` and `window_max.y` if `window_min.y > window_max.y`.
+ *
+ * @param[in] window_min lower-left (x, y) coordinate of the query window
+ * @param[in] window_max upper-right (x, y) coordinate of the query window
```

Comment:
There isn't a uniform coordinate system in cuspatial so calling it lower-left and upper-right could be confusing. I think it's best to refer to them as "one of the opposite corners of the query window". Analogous to opencv: https://docs.opencv.org/4.x/d6/d6e/group__imgproc__draw.html#ga07d2f74cadcf8e305e810ce8eed13bc9

Reply:
Should we specify that these are cartesian coordinates, in this context?

Reply:
I like this suggestion. So the names `min` and `max` are not appropriate either...